fpga/src/main/scala/arty/FPGAChip.scala

```diff
@@ -0,0 +1,193 @@
+// See LICENSE for license details.
+package sifive.freedom.everywhere.e300artydevkit
+
+import Chisel._
+import chisel3.core.{attach}
+import chisel3.experimental.{withClockAndReset}
+
+import freechips.rocketchip.config._
+import freechips.rocketchip.diplomacy.{LazyModule}
+
+import sifive.blocks.devices.gpio._
+import sifive.blocks.devices.spi._
+
+import sifive.fpgashells.shell.xilinx.artyshell.{ArtyShell}
+import sifive.fpgashells.ip.xilinx.{IBUFG, IOBUF, PULLUP, PowerOnResetFPGAOnly}
+
+//-------------------------------------------------------------------------
+// E300ArtyDevKitFPGAChip
+//-------------------------------------------------------------------------
+
+class E300ArtyDevKitFPGAChip(implicit override val p: Parameters) extends ArtyShell {
```

Comment:
This might be a total nit, but I think FireSim uses FPGATop for its top level FPGA name. I'm wondering whether we want any type of naming convention consistency between the two? @davidbiancolin @sagark , thoughts?

Reply:
This class should eventually be similar to the SW sim `TestHarness` class, which is also similar to firesim's `FireSim` class. Perhaps `TestHarness`, `FireSimHarness`, `FPGAHarness`?